pkg/scheduler/framework/plugins/volumebinding/volume_binding_test.go

```diff
@@ -267,19 +264,25 @@ func TestVolumeBinding(t *testing.T) {
 			for _, pv := range item.pvs {
 				client.CoreV1().PersistentVolumes().Create(ctx, pv, metav1.CreateOptions{})
 			}
+
+			t.Log("Start informer factory after initialization")
+			informerFactory.Start(ctx.Done())
+
 			caches := informerFactory.WaitForCacheSync(ctx.Done())
 			for _, synced := range caches {
 				if !synced {
 					t.Errorf("error waiting for informer cache sync")
 				}
 			}
```

Comment:
Just for curiosity, is this logic necessary? IMO a common pattern is just:

```go
informerFactory.Start(ctx.Done())
informerFactory.WaitForCacheSync(ctx.Done())
```

Reply:
practically, not necessary. In the current implementation, this function returns with the cache not synced only if the `ctx.Done()` is closed early, which will not happen in our testing.

However, I think it's good practice to check all returned results and fail if something is not expected. This may seem redundant sometimes, but it does not assume anything. For example, controllers in kube-controller-manager may exit if the informers they care can't be synced;

https://github.com/kubernetes/kubernetes/blob/2b9837fdcda0163712a439b71211e34a3f27bd34/pkg/controller/volume/persistentvolume/pv_controller_base.go#L306-L308

Reply:
I removed this logic since it's not necessary in our testing. The code is more concise now.